portal-api/api-tests/deal-versioning.api-test.js

```diff
@@ -0,0 +1,180 @@
+    monthsOfCover: null,
+    details: null,
+    detailsOther: null,
+    currency: null,
+    value: null,
+    coverPercentage: null,
+    interestPercentage: null,
+    paymentType: null,
+    createdAt: expect.any(Number),
+    updatedAt: expect.any(Number),
+    ukefExposure: 0,
+    guaranteeFee: 0,
+    submittedAsIssuedDate: null,
+    ukefFacilityId: null,
+    dayCountBasis: null,
+    feeType: null,
+    feeFrequency: null,
+    coverDateConfirmed: null,
+    canResubmitIssuedFacilities: null,
+    issueDate: null,
+    unissuedToIssuedByMaker: expect.any(Object),
+    hasBeenIssuedAndAcknowledged: null,
+    auditRecord: generateParsedMockPortalUserAuditDatabaseRecord(makerId),
+  },
+  validation: {
+    required: ['monthsOfCover', 'details', 'currency', 'value', 'coverPercentage', 'interestPercentage', 'feeType', 'feeFrequency', 'dayCountBasis'],
+  },
+});
+
+describe('GEF deal versioning', () => {
+  let aMaker;
+  let anAdmin;
+  let testUsers;
+
+  beforeAll(async () => {
+    testUsers = await testUserCache.initialise(app);
+    aMaker = testUsers().withRole(MAKER).one();
+    anAdmin = testUsers().withRole(ADMIN).one();
+  });
+
+  beforeEach(async () => {
+    await databaseHelper.wipe([MONGO_DB_COLLECTIONS.DEALS, MONGO_DB_COLLECTIONS.FACILITIES, MONGO_DB_COLLECTIONS.ELIGIBILITY_CRITERIA]);
+    await as(anAdmin).post(mockEligibilityCriteriaVersion).to('/v1/gef/eligibility-criteria');
+  });
+
+  describe(`when GEF_DEAL_VERSION set to '0'`, () => {
+    beforeAll(() => {
+      process.env.GEF_DEAL_VERSION = '0';
+    });
+
+    afterAll(() => {
+      process.env = originalEnv;
+    });
+
+    it('returns the expected version 0 application upon creation', async () => {
+      const { body } = await as(aMaker).post(getVersion0ApplicationToSubmit()).to(gefApplicationsUrl);
+
+      expect(body).toEqual(expectMongoId(getVersion0Application(aMaker._id)));
+    });
+
+    it('returns the expected version 0 facility on creation', async () => {
+      const { body: dealBody } = await as(aMaker).post(getVersion0ApplicationToSubmit()).to(gefApplicationsUrl);
+      const { body } = await as(aMaker).post({ dealId: dealBody._id, type: FACILITY_TYPE.CASH, hasBeenIssued: false }).to(gefFacilitiesUrl);
+
+      expect(body).toEqual(getVersion0Facility(dealBody._id, aMaker._id));
+    });
+
+    it('returns 400 when adding a facility end date to a version 0 facility', async () => {
+      const { body: dealBody } = await as(aMaker).post(getVersion0ApplicationToSubmit()).to(gefApplicationsUrl);
+      const { body: facilityBody } = await as(aMaker).post({ dealId: dealBody._id, type: FACILITY_TYPE.CASH, hasBeenIssued: false }).to(gefFacilitiesUrl);
+      const { status, body } = await as(aMaker).put({ facilityEndDateExists: true }).to(`${gefFacilitiesUrl}/${facilityBody.details._id}`);
+
+      expect(status).toBe(400);
+      expect(body).toEqual({ status: 400, message: 'Cannot add facility end date to deal version 0' });
```

Comment:
love this, are we covering the opposite, ie v1 missing fac end date?

Reply:
Updates wont throw an error if the facility end date is missing - I'll add a test for 200 with a facility end date